pkg/resourceutils/updatereplicas.go

```diff
@@ -341,3 +345,94 @@ func getUpdatedReplicaCount(annotations map[string]string, activate bool, printF
+	configMaps, err := core.Instance().ListConfigMap(namespace, metav1.ListOptions{LabelSelector: "stash-cr"})
+	if err != nil {
+		util.CheckErr(err)
+		return
+	}
+	ruleset := resourcecollector.GetDefaultRuleSet()
+	// Create the CRs in the same namespace if those dont exist
+	for _, configMap := range configMaps.Items {
+		objBytes := []byte(configMap.Data["runtime-object"])
+		unstructuredObj := &unstructured.Unstructured{}
+		//err := json.Unmarshal(objBytes, &unstructuredObj)
+		err := unstructuredObj.UnmarshalJSON(objBytes)
+		if err != nil {
+			printFunc(fmt.Sprintf("Error converting string to Unstructured object: %s\n", err.Error()), "err")
+			continue
+		}
+		configClient, err := k8sdynamic.NewForConfig(config)
+		if err != nil {
+			util.CheckErr(err)
+			continue
+		}
+		//var client k8sdynamic.ResourceInterface
+		resource := &metav1.APIResource{
+			Name:       ruleset.Pluralize(strings.ToLower(unstructuredObj.GetKind())),
+			Namespaced: len(unstructuredObj.GetNamespace()) > 0,
+		}
+
+		var resourceClient k8sdynamic.ResourceInterface
+		if resource.Namespaced {
+			resourceClient = configClient.Resource(
+				unstructuredObj.GetObjectKind().GroupVersionKind().GroupVersion().WithResource(resource.Name)).Namespace(unstructuredObj.GetNamespace())
+		} else {
+			resourceClient = configClient.Resource(
+				unstructuredObj.GetObjectKind().GroupVersionKind().GroupVersion().WithResource(resource.Name))
+		}
+		//resourceInterface := configClient.Resource()
+		//client = resourceInterface.Namespace(namespace)
+
+		// Create the CR
+		_, err = resourceClient.Create(context.TODO(), unstructuredObj, metav1.CreateOptions{})
+		if err != nil && !errors.IsAlreadyExists(err) {
+			printFunc(fmt.Sprintf("Error creating resource %s/%s from stashed configmap : %v\n", unstructuredObj.GetKind(), unstructuredObj.GetName(), err), "err")
+			util.CheckErr(err)
+			continue
+		}
+
+		// Get the CR
+		newParentResourceUnstructured, err := resourceClient.Get(context.TODO(), unstructuredObj.GetName(), metav1.GetOptions{})
+		if err != nil {
+			util.CheckErr(err)
+			continue
+		}
+
+		curResourceKey := fmt.Sprintf("%s-%s", unstructuredObj.GetName(), unstructuredObj.GetNamespace())
+		newOwnerUIDMapping[curResourceKey] = newParentResourceUnstructured.GetUID()
+		printFunc(fmt.Sprintf("Successfully created the CRs from the stashed configmaps %s/%s", unstructuredObj.GetKind(), unstructuredObj.GetName()), "out")
+	}
+
+	// Find PVCs in the namespace which are owned by previous CR, need to modify the owner reference to new CRs for those
+	pvcList, err := core.Instance().GetPersistentVolumeClaims(namespace, nil)
+	if err != nil {
+		util.CheckErr(err)
+		return
+	}
+
+	for _, pvc := range pvcList.Items {
+		destPvcOwnersList := make([]metav1.OwnerReference, 0)
+		needUpdate := false
+		for _, pvcOwner := range pvc.GetOwnerReferences() {
+			if newOwnerUID, ok := newOwnerUIDMapping[fmt.Sprintf("%s-%s", pvcOwner.Name, pvcOwner.Kind)]; ok {
```

Comment:
We are putting(name,namespace) as the key in newOwnerUIDMapping. But while fetching we are using (name, kind)

Reply:
good catch, fixed it now. thanks.